### PR TITLE
Add generate_sidecar as config option

### PR DESCRIPTION
**Pull request by g13ydson**
### Summary

Add `config.view_component.generate_sidecar` as config option

### Other Information

possibly resolves this issue #1216

### CODE REVIEW

**Comment by boardfish** [comment]
I'm just wondering if there's a way to expose a layer of config for all generator options. There are other things like the parent class name that folks may want to set defaults for, so if we're going to introduce config for generator defaults, I think it should be resilient to new generator options. @Spone's got the smarts when it comes to generators, so let's see what they say. That's what I'm thinking, though.

**Comment by Spone** [comment]
Thanks for the PR! I'll review this in a few days, I'm off for holidays.

**Comment by g13ydson** [comment]
I set the default ruby version 3 to 3.0.3 as the CI was jumping to version 3.1 (I don't know why).
Some tests with Rails 6.1 + ruby 3.1 failed. Here we have an explanation: [https://gist.github.com/yahonda/2776d8d7b6ea7045359f38c10449937b](https://gist.github.com/yahonda/2776d8d7b6ea7045359f38c10449937b)

```
Run ruby/setup-ruby@v1
  with:
    ruby-version: 3
    bundler: default
    bundler-cache: false
    working-directory: .
    cache-version: 0
Modifying PATH
  Entries added to PATH to use selected Ruby:
    /opt/hostedtoolcache/Ruby/3.1.0/x64/bin
Downloading Ruby
  https://github.com/ruby/ruby-builder/releases/download/toolcache/ruby-3.1.0-ubuntu-20.04.tar.gz
```

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,12 +27,12 @@ jobs:
     strategy:
       matrix:
         rails_version: [5.2.5, 6.0.3.6, 6.1.3.1, main]
-        ruby_version: [2.5, 2.6, 2.7, 3.0]
+        ruby_version: [2.5, 2.6, 2.7, 3.0.3]
```
Comment (Spone) [inline .github/workflows/ci.yml:30]: Please revert these changes, as the issue will be solved separately, in #1224

**Comment by Spone** [comment]
@g13ydson Please make sure to run `bundle exec rake docs:build` to regenerate docs/api.md to include changes in YARD comments from this pull request.

File: lib/view_component/base.rb [hunk 2]
```diff
@@ -310,8 +310,15 @@ def content_evaluated?
...
-    mattr_accessor :component_parent_class,
-                   instance_writer: false
+    mattr_accessor :component_parent_class, instance_writer: false
+
+    # Always generate a component with a sidecar directory:
+    #
+    #     config.view_component.generate_sidecar = true
+    #
+    # Defaults to `false`.
+    #
```
Comment (Spone) [inline lib/view_component/base.rb:320]:
```suggestion
```

**Comment by Spone** [comment]
> I'm just wondering if there's a way to expose a layer of config for all generator options. There are other things like the parent class name that folks may want to set defaults for, so if we're going to introduce config for generator defaults, I think it should be resilient to new generator options. @Spone's got the smarts when it comes to generators, so let's see what they say. That's what I'm thinking, though.

I agree with @boardfish: now that the generators have quite a few options, it would be interesting to refactor this, in a separate PR. Who wants to pair with me on this?

**Review by Spone — APPROVED** [review]

**Comment by boardfish** [comment]
@Spone I'm up for pairing on that!

**Comment by Spone** [comment]
> @Spone I'm up for pairing on that!

Pointer to the issue where we will work on this: #1229
